determined-ai/determined
# feat: generate and use swagger typescript client [DET-3249 DET-3324 DET-3355]

## Description

depends on #680 

## Test Plan

- make sure we can build the npm module: a java step and a Node step. 
- make sure WebUI builds and we can use a generated endpoint in React (logout)
  - Click the logout to test logout and check that the cookie is invalid
- make sure CI builds in a reasonable time
- check and make sure types and API comments come through the generated TS code.


## Commentary (optional)







[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234
